src/LibraryManager.Contracts/FileHelpers.cs

```diff
@@ -52,21 +65,31 @@ public static async Task<bool> SafeWriteToFileAsync(string fileName, Stream sour
                 // Clean up temp file if we didn't move it to the desination file successfully
                 if (!result)
                 {
-                    try
-                    {
-                        DeleteFiles(new string[] { tempFileName });
-                    }
-                    catch
-                    {
-                        Debug.Fail($"Could not clean up temporary file {tempFileName}");
-                        // Don't fail the operation if we couldn't clean up temporary file
-                    }
+                    SafeDeleteTempFile(tempFileName);
+                }
+
+                if (tempFileName != originalTempFileName)
+                {
+                    SafeDeleteTempFile(originalTempFileName);
                 }
             }
 
             return result;
         }
 
+        private static void SafeDeleteTempFile(string tempFileName)
```

Comment:
I don't see why Temp is relevant in this.  Why not just delete the file directly?  It should be apparent from the usage that you're deleting the temp files
```
DeleteFileFromDisk(tempFileName); <-- pretty clear that you're deleting the temp file
```

If the Debug.Fail is the only value this adds: a) it doesn't do anything on non-Windows (where there aren't non-fatal Debug listeners by default), and b) why not move that to `DeleteFileFromDisk`, which already has a try-catch.  This one puts us (I think) 3 levels deep of try-catching around the actual call to `File.Delete()`... I don't think it will actually have anything to catch.

Reply:
That's fair, didn't realize we already had DeleteFile that did try/catch. I'll leave the try/catch in DeleteFile though just in case. In my experience, I/O operations **always** throw, earlier or later.